### ISSUE TITLE
Loss Oscillating - Confirm Training Details

### ISSUE THREAD

**Issue by mitkina** [nutonomy/nuscenes-devkit#578]
Hello,

I am training CoverNet, and finding that the training loss is oscillating. I am using the following parameters. Could you confirm if these match with the original paper implementation?

learning_rate: 1e-4
optimizer: SGD
momentum: 0.9
batch_size: 16 (due to memory limitations - I know the original paper used 64)
epochs: 60 (the original paper used 20-30?)

Thanks!

**Comment by emwolff** [comment]
@mitkina Please see https://github.com/nutonomy/nuscenes-devkit/issues/543

Batch size: 64
Epochs: 20-30 for nuScenes 6 sec. Convergence was around 9.
Optimizer: SGD with momentum of 0.9 and weight decay of 5e-4.

Perhaps the weight decay is the issue? Or maybe the smaller batch size would benefit from a smaller learning rate as the gradient is a bit noisier.

**Comment by mitkina** [comment]
Thanks! I was looking for confirmation on learning rate? Also, by weight decay, do you mean learning rate decay? I saw the other issue before posting mine.

**Comment by holger-motional** [comment]
@mitkina For weight decay, please refer to https://en.wikipedia.org/wiki/Convolutional_neural_network#Explicit. It's a commonly use regularization technique.

**Comment by mitkina** [comment]
Got it thanks! Can you confirm the learning rate of 1e-4? I think I just needed to train for longer though, because it looks better now.

**Comment by emwolff** [comment]
@mitkina Yes, confirmed 1e-4 learning rate. You might need to adjust as you have a small batch size, which means that the gradient estimation will be noisier.
